bittensor/utils/version.py

```diff
@@ -0,0 +1,89 @@
+from typing import Optional
+from pathlib import Path
+import time
+
+import bittensor
+import requests
+
+VERSION_CHECK_THRESHOLD = 86400
+
+
+def _get_version_file_path() -> Path:
+    return Path.home() / ".bittensor" / ".version"
+
+
+def _get_version_from_file(version_file: Path) -> Optional[str]:
+    if not version_file.exists():
+        bittensor.logging.debug("No bitensor version file found")
+        return None
+
+    mtime = version_file.stat().st_mtime
+    bittensor.logging.debug(f"Found version file, last modified: {mtime}")
+    diff = time.time() - mtime
+
+    if diff >= VERSION_CHECK_THRESHOLD:
+        bittensor.logging.debug("Version file expired")
+        return None
+
+    try:
+        return version_file.read_text()
+    except Exception as e:
+        bittensor.logging.error(f"Failed to read version file: {e}")
+        return None
+
+
+def _get_version_from_pypi(timeout: int = 15) -> str:
+    try:
+        bittensor.logging.debug(
+            f"Checking latest Bittensor version at: {bittensor.__pipaddress__}"
+        )
+        response = requests.get(bittensor.__pipaddress__, timeout=timeout)
+        latest_version = response.json()["info"]["version"]
+
+    except requests.exceptions.Timeout:
+        bittensor.logging.error("Version check failed due to timeout")
+    except requests.exceptions.RequestException as e:
+        bittensor.logging.error(f"Version check failed due to request failure: {e}")
+    else:
+        return latest_version
+
+    raise
+
+
+def get_latest_version(timeout: int = 15) -> str:
+    version_file = _get_version_file_path()
+    latest_version = _get_version_from_file(version_file)
+
+    if not latest_version:
+        try:
+            latest_version = _get_version_from_pypi(timeout)
+        except Exception:
+            return None
+        try:
+            version_file.write_text(latest_version)
+        except Exception as e:
+            bittensor.logging.error(f"Failed to write version file: {e}")
+
+    return latest_version
+
+
+def version_checking(timeout: int = 15):
+    try:
+        latest_version = get_latest_version(timeout)
+    except Exception:
+        return
+
+    version_split = latest_version.split(".")
+    latest_version_as_int = (
+        (100 * int(version_split[0]))
+        + (10 * int(version_split[1]))
+        + (1 * int(version_split[2]))
+    )
```

Comment:
Will it break on versions like `6.0.0a1`?
And on versions like `6.0.15` vs `6.1.5`?

Reply:
That's a good point. I actually just moved this piece of code assuming it's robust. Will think of something better here

Reply:
I know that you just moved the code, but we can't ignore such obvious mistakes anyway 🙂